src/main/resources/META-INF/rewrite/java-version-11.yml

```diff
@@ -70,7 +70,7 @@ recipeList:
   - org.openrewrite.java.migrate.RemovedPolicy
   - org.openrewrite.java.migrate.ReferenceCloneMethod
   - org.openrewrite.java.migrate.ThreadStopDestroy
-
+  - org.openrewrite.java.migrate.ReplaceAWTGetPeerMethod
```

Comment:
Here you likely need to provide the missing arguments three times, for Component, Font and MenuComponent, as per the test failure:
```
    Missing{property='ReplaceAWTGetPeerMethod.className', value='null', message='is required'},
    Missing{property='ReplaceAWTGetPeerMethod.methodPatternGetPeer', value='null', message='is required'},
    Missing{property='ReplaceAWTGetPeerMethod.methodUpdateIsDisplayable', value='null', message='is required'},
    Missing{property='ReplaceAWTGetPeerMethod.methodUpdateIsLightweight', value='null', message='is required'}],
```

Reply:
@timtebeek I updated the java-version-11.yml I don't see that error anymore. I only updated the arguments once since we use the methodMatcher the Fonts, Components and Menu Component are also replaced.Here is my before and after code:
```
package com.acme.modres;

import java.awt.Button;
import java.awt.Canvas;
import java.awt.Checkbox;
import java.awt.Choice;
import java.awt.Component;
import java.awt.Container;
import java.awt.peer.ComponentPeer;// 1 eclipse flag

import javax.swing.plaf.FontUIResource;

import java.awt.Font;
import java.awt.Label;
import java.awt.List;
import java.awt.MenuBar;
import java.awt.MenuComponent;
import java.awt.MenuItem;
import java.awt.Scrollbar;
import java.awt.TextComponent;
import java.awt.dnd.peer.DropTargetPeer;
import java.awt.peer.LightweightPeer;

public class TestPeer {

	public void doX(Component component, Font font, MenuComponent mc, TextComponent textComponent) {
		ComponentPeer peer = component.getPeer();
		if (component.isDisplayable()) {};
		
		if(font.getPeer() != null);{}
		if(mc.getPeer() != null);{}
		
		Button button = new Button();
		Canvas canvas = new Canvas();
		Checkbox checkbox = new Checkbox();
		Choice choice = new Choice();
		Container container = new Container();
		Label label = new Label();
		List list = new List();
		Scrollbar scrollbar = new Scrollbar();
		FontUIResource fontUIResource = new FontUIResource(font);
		MenuBar menuBar = new MenuBar();
		MenuItem menuItem = new MenuItem();
		
		if(button.getPeer() != null);{}
		if(canvas.getPeer() != null);{}
		if(checkbox.getPeer() != null);{}
		if(choice.getPeer() != null);{}
		if(container.getPeer() != null);{}
		if(label.getPeer() != null);{}
		if(list.getPeer() != null);{}
		if(scrollbar.getPeer() != null);{}
		if(fontUIResource.getPeer() != null);{}
		if(menuBar.getPeer() != null);{}
		if(menuItem.getPeer() != null);{}
	}
	
	public void addNotify(ComponentPeer peer, Component component) { 
        for (Component c = component; 
             c != null && c.getPeer() instanceof java.awt.peer.LightweightPeer; c = c.getParent()) { 
        } 
 
        if ( component.getPeer() instanceof DropTargetPeer) { 
            ((DropTargetPeer)peer).addDropTarget(null); 
        } 
        
        if ( component.getPeer() instanceof java.awt.dnd.peer.DragSourceContextPeer) { 
            ((java.awt.dnd.peer.DragSourceContextPeer)peer).transferablesFlavorsChanged(); 
        } 
         if ( component.getPeer() instanceof LightweightPeer) { 
            ((java.awt.dnd.peer.DragSourceContextPeer)peer).transferablesFlavorsChanged(); 
        } 
        if ( component.getPeer() instanceof java.awt.peer.LightweightPeer) { 
            ((java.awt.dnd.peer.DragSourceContextPeer)peer).transferablesFlavorsChanged(); 
        } 
        
        if (component.isDisplayable()) {};
        if (component.isLightweight()) {};
    } 
}
```
After:
```
package com.acme.modres;

import java.awt.Button;
import java.awt.Canvas;
import java.awt.Checkbox;
import java.awt.Choice;
import java.awt.Component;
import java.awt.Container;
import java.awt.peer.ComponentPeer;// 1 eclipse flag

import javax.swing.plaf.FontUIResource;

import java.awt.Font;
import java.awt.Label;
import java.awt.List;
import java.awt.MenuBar;
import java.awt.MenuComponent;
import java.awt.MenuItem;
import java.awt.Scrollbar;
import java.awt.TextComponent;
import java.awt.dnd.peer.DropTargetPeer;
import java.awt.peer.LightweightPeer;

public class TestPeer {

	public void doX(Component component, Font font, MenuComponent mc, TextComponent textComponent) {
		ComponentPeer peer = component.getPeer();
		if (component.isDisplayable()) {};
		
		if(font.isDisplayable());{}
		if(mc.isDisplayable());{}
		
		Button button = new Button();
		Canvas canvas = new Canvas();
		Checkbox checkbox = new Checkbox();
		Choice choice = new Choice();
		Container container = new Container();
		Label label = new Label();
		List list = new List();
		Scrollbar scrollbar = new Scrollbar();
		FontUIResource fontUIResource = new FontUIResource(font);
		MenuBar menuBar = new MenuBar();
		MenuItem menuItem = new MenuItem();
		
		if(button.isDisplayable());{}
		if(canvas.isDisplayable());{}
		if(checkbox.isDisplayable());{}
		if(choice.isDisplayable());{}
		if(container.isDisplayable());{}
		if(label.isDisplayable());{}
		if(list.isDisplayable());{}
		if(scrollbar.isDisplayable());{}
		if(fontUIResource.isDisplayable());{}
		if(menuBar.isDisplayable());{}
		if(menuItem.isDisplayable());{}
	}
	
	public void addNotify(ComponentPeer peer, Component component) { 
        for (Component c = component; 
             c != null && c.isLightweight(); c = c.getParent()) { 
        } 
 
        if ( component.getPeer() instanceof DropTargetPeer) { 
            ((DropTargetPeer)peer).addDropTarget(null); 
        } 
        
        if ( component.getPeer() instanceof java.awt.dnd.peer.DragSourceContextPeer) { 
            ((java.awt.dnd.peer.DragSourceContextPeer)peer).transferablesFlavorsChanged(); 
        } 
         if ( component.isLightweight()) { 
            ((java.awt.dnd.peer.DragSourceContextPeer)peer).transferablesFlavorsChanged(); 
        } 
        if ( component.isLightweight()) { 
            ((java.awt.dnd.peer.DragSourceContextPeer)peer).transferablesFlavorsChanged(); 
        } 
        
        if (component.isDisplayable()) {};
        if (component.isLightweight()) {};
    } 
}
```